extensions/amp-base-carousel/1.0/arrow.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Preact from '../../../src/preact';
+import * as styles from './base-carousel.css';
+import {dict} from '../../../src/utils/object';
+
+/**
+ * @param {!JsonObject} props
+ * @return {PreactDef.Renderable}
+ */
+export function Arrow(props) {
+  const {
+    'customArrow': customArrow,
+    'dir': dir,
+    'onClick': onClick,
+    'disabled': disabled,
+  } = props;
+  const button = customArrow
+    ? customArrow
+    : renderDefaultArrow(dict({'dir': dir}));
+  const {
+    'children': children,
+    'disabled': customDisabled,
+    'onCustomClick': onCustomClick,
+    ...rest
+  } = button.props;
+  const isDisabled = disabled || customDisabled;
+  const handleClick = (e) => {
+    if (onCustomClick) {
+      onCustomClick(e);
```

Comment:
The arrow's props control the carousel, and the button props could be passed in by a publisher via their custom button. We do the same thing in amp-selector:
https://github.com/ampproject/amphtml/blob/b504ae08c211295f3eedaa3d387333ca635f350d/extensions/amp-selector/1.0/selector.js#L105-L113

The reason we pass in the arrow `onClick` from `BaseCarousel` instead of some `set` is discussed more [in this thread](https://github.com/ampproject/amphtml/pull/28704#discussion_r451642026).

Reply:
Thank you for your feedback though -- I've decided we don't actually need `onClick` to have that prop name on `Arrow` and have renamed it to a less confusing, self-describing `advance` property.